### ISSUE TITLE
[studio] unlock a Groovy script failed with invalid parameter exception

### ISSUE THREAD

**Issue by phuongnq** [craftercms/craftercms#6002]
### Duplicates

- [X] I have searched the existing issues

### Latest version

- [X] The issue is in the latest released 4.0.x
- [ ] The issue is in the latest released 3.1.x

### Describe the issue

Error while unlocking a Groovy script:

```
[ERROR] 2023-03-31T09:27:51,686 [http-nio-8080-exec-8] [v2.ExceptionHandlers] | API endpoint 'http://localhost:8080/studio/api/2/content/item_unlock_by_path' failed with response 'ApiResponse{code=1001, message='Invalid parameter(s)', remedialAction='Check API and make sure you're sending the correct parameters', documentationUrl=''}' 
org.springframework.web.bind.MethodArgumentNotValidException: Validation failed for argument [0] in public org.craftercms.studio.model.rest.ResponseBody org.craftercms.studio.controller.rest.v2.ContentController.itemUnlockByPath(org.craftercms.studio.model.rest.content.UnlockItemByPathRequest) throws org.craftercms.studio.api.v1.exception.ContentNotFoundException,org.craftercms.studio.api.v2.exception.content.ContentAlreadyUnlockedException,org.craftercms.studio.api.v1.exception.SiteNotFoundException: [Field error in object 'unlockItemByPathRequest' on field 'path': rejected value [/scripts/rest/site-map.get.groovy]; codes [EsapiValidatedParam.unlockItemByPathRequest.path,EsapiValidatedParam.path,EsapiValidatedParam.java.lang.String,EsapiValidatedParam]; arguments [org.springframework.context.support.DefaultMessageSourceResolvable: codes [unlockItemByPathRequest.path,path]; arguments []; default message [path],CONTENT_PATH_WRITE]; default message [validation.error.esapi.CONTENT_PATH_WRITE]] 
```

### Steps to reproduce

Steps:

1. Create a site from Editorial BP
2. Click to open a REST script
3. Close it with the x button
4. Notice that the item is locked and has an error in catalina.out
5. Try unlocking the item, see the same error

### Screenshots and/or videos

![2023-03-31_09-56](https://user-images.githubusercontent.com/2996543/229080539-98029281-7d3d-4d41-bfa5-b6920e531650.png)

**Comment by phuongnq** [comment]
@sumerjabri Two PRs need to be merged at the same time. Here is my update on these:

* Use `EsapiValidationType.ASSET_PATH_WRITE` for unlock API since this regex supports multiple `.`
* Update `ASSET_PATH_WRITE` to accept capital characters since some Groovy scripts could have capital characters for Groovy classes. Ex: in the Editorial BP `ProfileUtils.groovy`, `SearchHelper.groovy`, etc

**Comment by sumerjabri** [comment]
CC @yacdaniel we missed this one. Please add to the automated tests.

**Comment by yacdaniel** [comment]
Verified fixed in `4.1.0-SNAPSHOT`
